tests/unittests/tests-uri_parser/tests-uri_parser.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (C) 2020 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+
+#include <stdio.h>
+#include "embUnit.h"
+
+#include "uri_parser.h"
+
+#include "unittests-constants.h"
+#include "tests-uri_parser.h"
+
+#define VEC(u, s, us, h, po, pa, q, e)                                  \
+    { .uri = u, .scheme = s, .userinfo = us, .host = h, .port = po,     \
+      .path = pa, .query = q, .expected = e}
+
+#define VEC_CHECK(comp)                                                 \
+    do {                                                                \
+        if (ures.comp == NULL) {                                        \
+            TEST_ASSERT(validate_uris[i].comp[0] == '\0');              \
+        }                                                               \
+        else {                                                          \
+            TEST_ASSERT_EQUAL_INT(strlen(validate_uris[i].comp),        \
+                                  ures.comp##_len);                     \
+            TEST_ASSERT_EQUAL_INT(0,                                    \
+                                  memcmp(ures.comp,                     \
+                                         validate_uris[i].comp,         \
+                                         strlen(validate_uris[i].comp))); \
+        }                                                               \
+    } while (0)
+
+typedef struct {
+    char uri[64];
+    char scheme[8];
+    char userinfo[16];
+    char host[16];
+    char port[32];
+    char path[48];
+    char query[32];
+    int expected;
+} validate_t;
+
+/*
+  VEC(uri_to_parse,
+      scheme, userinfo, host, port,
+      path, query, expected return value)
+*/
+const validate_t validate_uris[24] = {
```

Comment:
```suggestion
static const validate_t validate_uris[24] = {
```

?

Reply:
yup, amended the small change. thanks :)